### PR TITLE
added isMounted to check if the component has been unmounted before s…

### LINKED ISSUE #3041: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
**Describe the bug**
I am using Amplify in a React-Native project. After signing in with Facebook or Google, it will show this warning. I have removed the listener to avoid memory leaks. And I also have tried to add a set of "_isMounted" states as some of the instructions say, but either of them worked. 

**To Reproduce**
Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.

**Expected behavior**

**Screenshots**
image

**Sample code**
image

### PR DESCRIPTION
**Pull request by inh556**
…etState, related to issue #598

*Issue #, if available:*
issue #598
fixes #3041 
*Description of changes:*
added _isMounted states to check if the component has been unmounted before setState

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react-native/src/Auth/withOAuth.js [hunk 1]
```diff
@@ -49,15 +49,19 @@ export default (Comp) => {
...
         componentDidMount() {
+            this._isMounted = true;
             Auth.currentAuthenticatedUser().then(user => {
                 this.setState({ user })
             }).catch(error => {
                 logger.debug(error);
-
-                this.setState({ user: null });
+                if (this.this._isMounted) {
+                    this.setState({ user: null });
```
Comment (alebiavati) [inline packages/aws-amplify-react-native/src/Auth/withOAuth.js]: @inh556 I don't think this line is the problem you are facing. Based on our conversations on #3041, I thought that the `setState` that causes the warning is on line 72:

https://github.com/aws-amplify/amplify-js/blob/fb826d09cfd962af8ccd4660d15da38552756172/packages/aws-amplify-react-native/src/Auth/withOAuth.js#L72

Comment (inh556) [reply packages/aws-amplify-react-native/src/Auth/withOAuth.js]: @robbevan I am so sorry. I just corrected it to the right place.

File: packages/aws-amplify-react-native/src/Auth/withOAuth.js [hunk 2]
```diff
@@ -49,15 +49,17 @@ export default (Comp) => {
...
+                logger.debug(error)；
                 this.setState({ user: null });
             });
         }
-
+        componentWillUnmount() {
+            this._isMounted = false;
+        }
         onHubCapsule(capsule) {
             // The Auth module will emit events when user signs in, signs out, etc
```
Comment (powerful23) [inline packages/aws-amplify-react-native/src/Auth/withOAuth.js]: You can add this check here so if the component is not mounted, then just directly return.

Comment (inh556) [reply packages/aws-amplify-react-native/src/Auth/withOAuth.js]: @powerful23 Hi, I corrected the typo and move the check to the componentWillUnmount method, please check if this makes it work. Thank you.

File: packages/aws-amplify-react-native/src/Auth/withOAuth.js [hunk 3]
```diff
@@ -49,15 +49,18 @@ export default (Comp) => {
...
             });
         }
-
+        componentWillUnmount() {
+            this._isMounted = false;
+            return;
+        }
         onHubCapsule(capsule) {
             // The Auth module will emit events when user signs in, signs out, etc
             const { channel, payload } = capsule;
```
Comment (powerful23) [inline packages/aws-amplify-react-native/src/Auth/withOAuth.js]:
```suggestion
            if (!this._isMounted) return;
            const { channel, payload } = capsule;
```

**Pull request merged**